added purchase url parser

aliseyalvi opened the pull request:
### What does this PR?
This PR adds support for parsing purchase urls and navigate user to account boost screen when scanning a valid purchase url

### Issue number
fixes #2345 

### Screenshots/Video

Uploading Screenrecorder-2022-06-09-20-42-49-64.mp4…

noumantahir commented on src/utils/deepLinkParser.ts:
```diff
@@ -80,6 +81,18 @@ export const deepLinkParser = async (url, currentAccount) => {
+    if(type && type === 'boost'){
+      routeName = ROUTES.SCREENS.ACCOUNT_BOOST;
+      params = {
+        username,
```
accountBoostScreen is not reading navigation parameters, so this username is being discarded... read it in accountBoostScreen and pass it to InAppPurchaseContainer. The prop should be optional but if preset should override the currentAccount.name already being used in InAppPurchases.

references
https://github.com/ecency/ecency-mobile/blob/d1fa36330399c05048cd546723fefc719d690f10/src/containers/inAppPurchaseContainer.ts#L84

https://github.com/ecency/ecency-mobile/blob/d1fa36330399c05048cd546723fefc719d690f10/src/screens/accountBoost/screen/accountBoostScreen.js#L27

aliseyalvi commented:
@noumantahir  updated.

- added username param in purchase screen
- added user avatar in purchase screen

![image](https://user-images.githubusercontent.com/48380998/173054672-25e130b9-ae31-4920-899b-e81e2a7bb319.png)

aliseyalvi commented:
updated: 
![image](https://user-images.githubusercontent.com/48380998/173058283-d4479a42-9f5e-48a2-9df6-a718ee885037.png)

noumantahir commented:
Fine Tuned UI for account boost screen
![Screenshot 2022-06-10 at 5 35 59 PM](https://user-images.githubusercontent.com/6298342/173065361-8c4bceac-ec89-4d26-93c6-977284f82559.png)

noumantahir commented:
@feruzm you may test and merge this.... QR code for testing....
![qrcode (1)](https://user-images.githubusercontent.com/6298342/173066403-f4046cd0-4993-4473-a8ab-57fb3eb1bde0.png)
https://ecency.com/purchase?type=boost&username=demo.com

feruzm reviewed (approved):
lgtm
